core/src/main/scala/org/apache/spark/executor/CoarseGrainedExecutorBackend.scala

```diff
@@ -283,7 +283,20 @@ private[spark] object CoarseGrainedExecutorBackend extends Logging {
         executorConf,
         new SecurityManager(executorConf),
         clientMode = true)
-      val driver = fetcher.setupEndpointRefByURI(arguments.driverUrl)
+
+      var driver: RpcEndpointRef = null
+      val nTries = 3
+      for (i <- 1 until nTries if driver == null) {
+        try {
+            driver = fetcher.setupEndpointRefByURI(arguments.driverUrl)
+          }
+        catch {
```

Comment:
The try-catch indent is off, but, this doesn't do anything? `i` can't reach `nTries`.

Reply:
Hi @srowen true, I'll fix it.